File: web/src/hooks/Theme.ts
import { useEmbeddedVariable } from "./Configuration";

export function useTheme() {
    return useEmbeddedVariable("theme-name");
}

export function usePrimaryColor() {
    return useEmbeddedVariable("theme-primarycolor");
}

export function useSecondaryColor() {
    return useEmbeddedVariable("theme-secondarycolor");
}

File: internal/configuration/schema/theme.go
package schema

// ThemeConfiguration represents the configuration related to styling.
type ThemeConfiguration struct {
	Name           string `mapstructure:"name"`
	PrimaryColor   string `mapstructure:"primary_color"`
	SecondaryColor string `mapstructure:"secondary_color"`
}

// DefaultThemeConfiguration represents default configuration parameters for styling.
var DefaultThemeConfiguration = ThemeConfiguration{
	Name:           "light",
	PrimaryColor:   "#1976d2",
	SecondaryColor: "#ffffff",
}
